### PR TITLE
Flow actions docs

### LINKED ISSUE #23279: Update documentation for configuration cache improvements in Gradle 8.1
Gradle 8.1 includes a lot of nice user facing improvements for configuration cache, but few of these are mentioned in the release notes or user manual.

### Expected Behavior

Users can more easily discover the configuration cache improvements in Gradle 8.1

### Current Behavior

Users need to keep a close watch on Gradle development to know about the improvements.

### To update

- Release notes
- Signing plugin chapter
- Configuration cache chapter

### PR DESCRIPTION
**Pull request by mlopatkin**
Fixes #23279

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test BD

**Comment by mlopatkin** [comment]
Rendered [Dataflow Actions](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/63306810:id/distributions/gradle-8.1-docs.zip!/gradle-8.1-20230316210319%2B0000/docs/userguide/dataflow_actions.html) page

File: subprojects/docs/src/docs/userguide/extending-gradle/dataflow_actions.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+include::{snippetsPath}/dataflowActions/playSound/groovy/plugin/src/main/java/org/gradle/sample/sound/SoundFeedbackPlugin.java[tag=flow-action]
+----
+====
+<1> Use service injection to obtain `FlowScope` and `FlowProviders` instances. They are available for project and settings plugins.
+<2> Use an appropriate scope to run your actions. As the name suggests, actions in the `always` scope are executed every time the build runs.
+<3> Specify the class that implements the action.
+<4> Use the spec argument to configure the action parameters.
+<5> A lifecycle event provider can be mapped into something else while preserving the action ordering.
+
+As a result, when you run the build, and it completes successfully, the action will play the "tada" sound. If the build fails at configuration or execution time, then you'll hear "sad-trombone" -- assuming, of course, that build configuration proceeds far enough for the action to be registered.
```
Comment (hythloda) [inline subprojects/docs/src/docs/userguide/extending-gradle/dataflow_actions.adoc:73]: nice

**Review by hythloda — APPROVED** [review]

File: subprojects/core-api/src/main/java/org/gradle/BuildListener.java [hunk 2]
```diff
@@ -67,6 +68,7 @@ default void beforeSettings(Settings settings) {}
      *
      * @param result The result of the build. Never null.
      * @deprecated This method is not supported when configuration caching is enabled.
+     * @see FlowProviders#getBuildWorkResult()
```
Comment (bamboo) [inline subprojects/core-api/src/main/java/org/gradle/BuildListener.java:71]: Nice 👍

File: subprojects/core-api/src/main/java/org/gradle/api/flow/FlowAction.java [hunk 3]
```diff
@@ -46,7 +46,8 @@
  * class FFPlay implements FlowAction&lt;FFPlay.Parameters&gt; {
  *
  *     interface Parameters extends FlowParameters {
- *         RegularFileProperty getMediaFile();
+ *         {@literal @}Input
```
Comment (bamboo) [inline subprojects/core-api/src/main/java/org/gradle/api/flow/FlowAction.java:49]: 👍

File: subprojects/core-api/src/main/java/org/gradle/api/flow/FlowAction.java [hunk 4]
```diff
@@ -61,7 +62,7 @@
  *         execOperations.exec(spec -&gt; {
  *             spec.commandLine(
  *                 "ffplay", "-nodisp", "-autoexit", "-hide_banner", "-loglevel", "quiet",
- *                 parameters.getMediaFile().get().getAsFile().getAbsolutePath()
+ *                 parameters.getMediaFile().get().getAbsolutePath()
```
Comment (bamboo) [inline subprojects/core-api/src/main/java/org/gradle/api/flow/FlowAction.java:65]: TIL 👍

File: subprojects/docs/src/docs/userguide/extending-gradle/dataflow_actions.adoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+These providers are intended for dataflow actions and provide additional ordering guarantees when used as inputs.
+The ordering also applies if you derive a provider from the event provider by, for example, calling `map` or `flatMap`.
+You can obtain these providers from the link:{javadocPath}/org/gradle/api/flow/FlowProviders.html[FlowProviders] class.
+
+WARNING: If you're not using a lifecycle event provider as an input to the dataflow action, then the exact timing when the action is executed is not defined and may change in the next version of Gradle.
+
+
+== Supplying the action for execution
+
+You should not create FlowAction objects manually.
```
Comment (bamboo) [inline subprojects/docs/src/docs/userguide/extending-gradle/dataflow_actions.adoc:56]:
```suggestion
You should not create `FlowAction` objects manually.
```

**Review by bamboo — APPROVED** [review]
Very nice!

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**